Deprecate EXPERIMENTAL_ env vars for code hostpots; introduce non-experimental ones

szegedi opened the pull request:
Deprecate `EXPERIMENTAL_` env vars for code hostpots; introduce non-experimental ones

### What does this PR do?
* Deprecates env vars `DD_PROFILING_EXPERIMENTAL_CODEHOTSPOTS_ENABLED` and `DD_PROFILING_EXPERIMENTAL_ENDPOINT_COLLECTION_ENABLED`. They are still supported, but non-experimental ones take precedence and the use of experimental ones results in logging of a deprecation warning.
* Introduces env vars `DD_PROFILING_CODEHOTSPOTS_ENABLED` and `DD_PROFILING_ENDPOINT_COLLECTION_ENABLED`

### Motivation
We are moving Code Hotspots feature to public beta.

nsavoire commented on packages/dd-trace/test/profiling/config.spec.js:
```diff
@@ -133,7 +133,7 @@ describe('config', () => {
   it('should support profiler config with DD_PROFILING_PROFILERS', () => {
     process.env = {
       DD_PROFILING_PROFILERS: 'wall',
-      DD_PROFILING_EXPERIMENTAL_CODEHOTSPOTS_ENABLED: '1',
```
Perhaps add a test that both options are supported and that DD_PROFILING_CODEHOTSPOTS_ENABLED overrides DD_PROFILING_EXPERIMENTAL_CODEHOTSPOTS_ENABLED

szegedi replied:
Done.

nsavoire replied:
Thanks for the changes !

nsavoire commented on packages/dd-trace/src/profiling/config.js:
```diff
@@ -71,11 +71,24 @@ class Config {
     )
     this.logger = ensureLogger(options.logger)
+    const logger = this.logger
+    function logExperimentalVarDeprecation(shortVarName) {
```
linter is complaining about formatting here

szegedi commented:
Okay, I added the requested tests.

While there, I also added a logic for handling the situation when endpoint profiling is enabled, but its prerequisite code hotspots, is not. Based on a Slack conversation with the team, the strategy for this is as follows:
* if code hotspots are not _explicitly_ disabled (so, no options or env vars are set at all) then implicitly turn code hotspots on, and log an info level message about it.
* if code hotspots are _explicitly_ disabled (by setting an option or env var to false) then turn endpoint profiling off, and log a warn level message about it. Debatable whether we should throw an error instead…
